src/flows/pipes/Notification/endpoints/PagerDuty/index.ts

```diff
@@ -21,14 +21,18 @@ export default register => {
       ['pagerduty', 'influxdata/influxdb/secrets']
         .map(i => `import "${i}"`)
         .join('\n'),
-    generateQuery: data => `task_data
+    generateQuery: (data, measurement) => `task_data
 	|> schema["fieldsAsCols"]()
-  |> set(key: "_notebook_link", value: "${window.location.href}")  
+  |> set(key: "_notebook_link", value: "${window.location.href}")
+  |> filter(fn: ${measurement})
   |> monitor["check"](
 		data: check,
 		messageFn: messageFn,
     crit: trigger,
 	)
+  |> filter(fn: trigger)
+  |> keep(columns: ["_value", "_time", "_measurement"])
```

Comment:
i dont understand these additions. what were you trying to address?

Reply:
Notification endpoints were being spammed. This is because the alerts were being triggered:
1. without the filter being applied at all
   * this is because alert.check() creates a separate task.
   * the `trigger` was only being applied within the alert check task.
   * it was NOT being applied in the current flux query. Therefore, monitor.notify() was always occurring (because `|> filter(fn: trigger)` was not occurring)

2. even once the filter is applied, there is several rows of data. Each row of data --> triggers a separate notification in the `monitor.notify()` 
   * use `keep()` to reduce cardinality of the data.
   * use `limit()` to then have a single row of data

A full description of what we are doing, and with the feedback from the flux team that this^^ query is correct, is available here:  https://influxdata.slack.com/archives/C0AD7FFS9/p1647972735067469